tests/test_optimizer.py

```diff
@@ -548,6 +548,39 @@ def test_scope_warning(self, logger):
             level="warning",
         )
 
+    def test_struct_type_annotation(self):
+        expression = annotate_types(parse_one("select struct(1 as col)"))
+        self.assertEqual(expression.expressions[0].type.this, exp.DataType.Type.STRUCT)
+        self.assertEqual(expression.expressions[0].expressions[0].type.this, exp.DataType.Type.INT)
```

Comment:
Not sure about this test, shouldn't we be checking whether the type is `STRUCT<INT>` here? Which would translate to matching the AST structure you've created with the `ColumnDef`s etc..

Reply:
Ditto about the other tests under this method

Reply:
done :D